libpromises/rlist.c

```diff
@@ -1044,6 +1059,56 @@ Rlist *RlistFromSplitString(const char *string, char sep)
     return liststart;
 }
 
+/**
+ * Splits the given string into lines. On Windows, both \n and \r\n newlines are
+ * detected. Escaped newlines are respected/ignored too.
+ *
+ * @return: an #Rlist where items are the individual lines **without** the
+ *          trailing newline character(s)
+ * @note: Free the result with RlistDestroy()
+ * @warning: This function doesn't work properly if @string uses "\r\n" newlines
+ *           and contains '\r' characters that are not part of any "\r\n"
+ *           sequence because it first splits @string on '\r'.
+ */
+Rlist *RlistFromStringSplitLines(const char *string)
+{
+#ifndef __MINGW32__
```

Comment:
Please make this into an if based on argument instead, so we can test more easily.

Reply:
Makes sense.